src/libraries/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs

```diff
@@ -98,9 +98,11 @@ internal static class DispatchProxyGenerator
         }
 
         // Unconditionally generates a new proxy type derived from 'baseType' and implements 'interfaceType'
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2062:UnrecognizedReflectionPattern",
+            Justification = "interfaceType is annotated as preserve All members, so any Types returned from GetInterfaces should be preserved as well once https://github.com/mono/linker/issues/1731 is fixed.")]
```

Comment:
Note I had to add this suppression back, with a link to https://github.com/mono/linker/issues/1731. The linker will have to preserve the whole interface chain if the interface is annotated with `All`.